viite-backend/base/src/test/scala/RoadPartSpec.scala

```diff
@@ -0,0 +1,105 @@
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(1,1)) shouldBe true
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(1,2)) shouldBe true
+
+    // different road
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(789,1)) shouldBe false
+    // unaddressed road or part
+    RoadPart(0,0).isAtSameRoadThan(RoadPart(0,0)) shouldBe false
+    RoadPart(0,0).isAtSameRoadThan(RoadPart(1,1)) shouldBe false
+    RoadPart(0,1).isAtSameRoadThan(RoadPart(1,1)) shouldBe false
+    RoadPart(1,0).isAtSameRoadThan(RoadPart(1,1)) shouldBe false
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(0,0)) shouldBe false
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(0,1)) shouldBe false
+    RoadPart(1,1).isAtSameRoadThan(RoadPart(1,0)) shouldBe false
+  }
+
+  test("Test RoadPart.isBefore: Returns true, if the compared RoadParts have same road number, and this RoadPArt has smaller part number than <i>other</i>, and both RoadParts are valid. Else false.") {
+    RoadPart(1,2).isBefore(RoadPart(1,3)) shouldBe true
+
+    // same road number, but not before
+    RoadPart(1,2).isBefore(RoadPart(1,2)) shouldBe false
+    RoadPart(1,2).isBefore(RoadPart(1,1)) shouldBe false
+    // different road number
+    RoadPart(1,2).isBefore(RoadPart(2,2)) shouldBe false
+    RoadPart(2,2).isBefore(RoadPart(1,2)) shouldBe false
+    // unaddressed road or part
+    RoadPart(0,1).isBefore(RoadPart(1,1)) shouldBe false
+    RoadPart(1,0).isBefore(RoadPart(1,1)) shouldBe false
+    RoadPart(1,1).isBefore(RoadPart(0,1)) shouldBe false
+    RoadPart(1,1).isBefore(RoadPart(1,0)) shouldBe false
+  }
+
+  test("Test RoadPart.isAfter: Returns true, if the compared RoadParts have same road number, and this RoadPArt has bigger  part number than <i>other</i>, and both RoadParts are valid. Else false.") {
+    RoadPart(1,2).isAfter(RoadPart(1,1)) shouldBe true
+
+    // same road number, but not after
+    RoadPart(1,2).isAfter(RoadPart(1,2)) shouldBe false
+    RoadPart(1,2).isAfter(RoadPart(1,3)) shouldBe false
+    // different road number
+    RoadPart(1,2).isAfter(RoadPart(2,2)) shouldBe false
+    RoadPart(2,2).isAfter(RoadPart(1,2)) shouldBe false
+    // unaddressed road or part
+    RoadPart(0,1).isAfter(RoadPart(1,1)) shouldBe false
+    RoadPart(1,0).isAfter(RoadPart(1,1)) shouldBe false
+    RoadPart(1,1).isAfter(RoadPart(0,1)) shouldBe false
+    RoadPart(1,1).isAfter(RoadPart(1,0)) shouldBe false
+  }
+
+  test("Test RoadPart.isUnaddressed: Returns true, if this RoadPart has both road number, and road part number zeroes. Else false.") {
+    RoadPart(0,1).isUnaddressed shouldBe true
+    RoadPart(1,0).isUnaddressed shouldBe true
+    RoadPart(0,0).isUnaddressed shouldBe true
+
+    // valid road and part
+    RoadPart(    1,  1).isUnaddressed shouldBe false
+    RoadPart(99999,999).isUnaddressed shouldBe false
+  }
+
+  test("Test RoadPart.isValid Returns true, if this RoadPart has both road number, and road part number greater than zeroes. Else false.") {
```

Comment:
Tässä myös testin kuvauksessa väitetään että molempien pitää olla nolla kun itse testin mukaan riittää että vain toinen numeroista on nolla

Reply:
Hmm, tässä en näe ongelmaa

Reply:
Oho joo oon varmaan silmät kierossa lukenu tätä :D Testin kuvaus on oikein 👍

mut zeroes -> zeros

Reply:
Oho, zeron monikkoa voi käyttää konteksti-sensitiivisesti:
https://www.wordhippo.com/what-is/the-plural-of/zero.html

Reply:
(Todettiin, että molemmat kirjoitusmuodot ok (am.eng/br.eng), annetaan olla.)